liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -401,39 +433,28 @@ public boolean recompileBuildFile(File buildFile, List<String> artifactPaths, Th
                 }
 
                 // monitoring Liberty plugin configuration changes in dev mode
-                if (!restartServer && !createServer) {
-                    Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(libertyPlugin, "create", log);
-                    Xpp3Dom currentConfig = ExecuteMojoUtil.getPluginGoalConfig(backupLibertyPlugin, "create", log);
-
-                    if (!Objects.equals(config.getChild("bootstrapProperties"),
-                            currentConfig.getChild("bootstrapProperties"))) {
-                        restartServer = true;
-                    } else if (!Objects.equals(config.getChild("bootstrapPropertiesFile"),
-                            currentConfig.getChild("bootstrapPropertiesFile"))) {
-                        restartServer = true;
-                    } else if (!Objects.equals(config.getChild("jvmOptions"),
-                            currentConfig.getChild("jvmOptions"))) {
-                        restartServer = true;
-                    } else if (!Objects.equals(config.getChild("jvmOptionsFile"),
-                            currentConfig.getChild("jvmOptionsFile"))) {
-                        restartServer = true;
-                    } else if (!Objects.equals(config.getChild("serverEnvFile"),
-                            currentConfig.getChild("serverEnvFile"))) {
-                        restartServer = true;
-                    } else if (!Objects.equals(config.getChild("configDirectory"),
-                            currentConfig.getChild("configDirectory"))) {
-                        restartServer = true;
-                        // restart dev mode
-                    } else {
-                        // assume createServer will call install-feature implicitly
-                        config = ExecuteMojoUtil.getPluginGoalConfig(libertyPlugin, "install-feature", log);
-                        currentConfig = ExecuteMojoUtil.getPluginGoalConfig(backupLibertyPlugin, "install-feature",
-                                log);
-                        if (!Objects.equals(config.getChild("features"), currentConfig.getChild("features"))) {
-                            installFeature = true;
+                Xpp3Dom config;
+                Xpp3Dom oldConfig;
+                if (!restartServer) {
+                    config = ExecuteMojoUtil.getPluginGoalConfig(libertyPlugin, "create", log);
+                    oldConfig = ExecuteMojoUtil.getPluginGoalConfig(backupLibertyPlugin, "create", log);
+                    if (!Objects.equals(config, oldConfig)) {
+                        createServer = true;
```

Comment:
This logic is different than before. Is it intentional? If you have to recreate the server, doesn't that automatically mean you need to restart the server? And before you only did this logic if neither restartServer or createServer were true. Now you only check restartServer above.

Reply:
Yes, it is intentional and the logic is to monitor the changes in Liberty plugin configuration.  There are situation that doesn't require a server restart but just a server recreate is enough.  for example, if `serverXmlFile` is change, we only need to call liberty:create again to update target server configuration and server restart is not required.